dtls.c

```diff
@@ -2801,7 +2866,7 @@ dtls_send_client_hello(dtls_context_t *ctx, dtls_peer_t *peer,
   psk = is_psk_supported(ctx);
   ecdsa = is_ecdsa_supported(ctx, 1);
 
-  cipher_size = 2 + ((ecdsa) ? 2 : 0) + ((psk) ? 2 : 0);
+  cipher_size = 2 + ((ecdsa) ? 4 : 0) + ((psk) ? 4 : 0);
```

Comment:
It would be good to have a comment here that the `4` refers to there being 2 algorithms that are going to be defined - to make it more obvious what needs to be done when another algorithm is added.
Perhaps `4` should be `2*2` to emphasize this further.

Reply:
I tried to improve the "length documentation" in a commit later 84f834a69eefefd4180edcda5a1b485cadb75b07 .
I will add also comment here.

Reply:
https://github.com/eclipse/tinydtls/commit/84f834a69eefefd4180edcda5a1b485cadb75b07 certainly helps, but does not specifically bring focus to this line for the supported suites that are going to be announced.
````
-  cipher_size = 2 + ((ecdsa) ? 2 : 0) + ((psk) ? 2 : 0);
+  /* Each announced cipher suite is 2 bytes long */
+  cipher_size = 2 + ((ecdsa) ? 2*2 : 0) + ((psk) ? 2*2 : 0);
````

Reply:
Yes, that commit doesn't address it, because this lines are removed on a [commit before](https://github.com/eclipse/tinydtls/commit/cb16f58ea036fa971d9e4eb91c67b9b4d4219552#diff-947f252e5a6242133f45e2963433822253886eb9a6bf4c12c7163ac8e90fe017R2906).

Therefore I think, it doesn't really pay-off to polish it more and then remove it.